src/app/products/manage-tax-groups/create-tax-group/create-tax-group.component.ts

```diff
@@ -0,0 +1,92 @@
+import { Component, OnInit } from '@angular/core';
+import { FormGroup, FormBuilder, Validators, FormArray } from '@angular/forms';
+import { ActivatedRoute, Router } from '@angular/router';
+import { ProductsService } from 'app/products/products.service';
+import { MatDialog } from '@angular/material';
+import { CancelDialogComponent } from 'app/shared/cancel-dialog/cancel-dialog.component';
+import { DatePipe } from '@angular/common';
+
+@Component({
+  selector: 'mifosx-create-tax-group',
+  templateUrl: './create-tax-group.component.html',
+  styleUrls: ['./create-tax-group.component.scss']
+})
+export class CreateTaxGroupComponent implements OnInit {
+  TaxGroupForm: FormGroup;
+    /** tax Component Data. */
+    taxComponentData: any;
+
+  constructor(private formBuilder: FormBuilder,
+              private route: ActivatedRoute,
+              private router: Router,
+              private datePipe: DatePipe,
+              public dialog: MatDialog,
+              private productService: ProductsService) {}
+
+  ngOnInit() {
+    // Create initial form
+    this.createTaxGroupForm();
+    this.productService.getTaxComponents().subscribe((data: any) => {
+      this.taxComponentData = data;
+    });
```

Comment:
Don't make direct calls to the API from the component, make it pass through a resolver instead as done in other components.